src/server/supplements/har/harTracer.ts

```diff
@@ -152,38 +155,50 @@ export class HarTracer {
         receive: -1
       },
     };
-    if (request.redirectedFrom()) {
-      const fromEntry = this._entries.get(request.redirectedFrom()!)!;
-      fromEntry.response.redirectURL = request.url();
-    }
+
     this._log.entries.push(harEntry);
     this._entries.set(request, harEntry);
   }
 
-  private _onResponse(page: Page, response: network.Response) {
+  private async _onRequestFinished(page: Page, request: network.Request) {
     const pageEntry = this._pageEntries.get(page)!;
-    const harEntry = this._entries.get(response.request())!;
+    const harEntry = this._entries.get(request)!;
     // Rewrite provisional headers with actual
-    const request = response.request();
     harEntry.request.headers = request.headers().map(header => ({ name: header.name, value: header.value }));
     harEntry.request.cookies = cookiesForHar(request.headerValue('cookie'), ';');
-    harEntry.request.postData = postDataForHar(request) || undefined;
+
+    const response = await request.response();
+
+    if (!response) return;
+
+    const httpVersion = response.protocol() ? response.protocol() : FALLBACK_HTTP_VERSION;
```

Comment:
There is already a fallback in the `protocol()` getter. Let's have it in one place (I prefer this one).

Reply:
@dgozman do you mean: "switch to my `FALLBACK_HTTP_VERSION`" or do you mean "keep current fallback"?

Reply:
I mean "make `response.protocol()` nullable" and keep `FALLBACK_HTTP_VERSION` here.

Reply:
aha, e.g.

```
 // Response class
  protocol(): string | undefined {
    return this._protocol;
  }
```

Reply:
ok, this was an easy fix. Pushed it already